MarvelApiClient/src/main/java/com/karumi/MarvelApiClient/model/CharactersQuery.java

```diff
@@ -0,0 +1,288 @@
+/*
+ *   Copyright (C) 2015 Karumi.
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.karumi.marvelapiclient.model;
+
+import com.karumi.marvelapiclient.utils.DateUtil;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CharactersQuery {
+
+  private static final String QUERY_NAME = "name";
+  private static final String QUERY_NAME_START_WITH = "nameStartsWith";
+  private static final String QUERY_MODIFIED_SINCE = "modifiedSince";
+  private static final String QUERY_COMICS = "comics";
+  private static final String QUERY_SERIES = "series";
+  private static final String QUERY_EVENTS = "events";
+  private static final String QUERY_STORIES = "stories";
+  private static final String QUERY_ORDER_BY = "orderBy";
+  private static final String QUERY_LIMIT = "limit";
+  private static final String QUERY_OFFSET = "offset";
+
+  private String name;
+  private String nameStartWith;
+  private String modifiedSince;
+  private String comics;
+  private String series;
+  private String events;
+  private String stories;
+  private String orderBy;
+  private int limit;
+  private int offset;
+
+  private CharactersQuery(String name, String nameStartWith, String modifiedSince, String comics,
+      String series, String events, String stories, String orderBy, int limit, int offset) {
+    this.name = name;
+    this.nameStartWith = nameStartWith;
+    this.modifiedSince = modifiedSince;
+    this.comics = comics;
+    this.series = series;
+    this.events = events;
+    this.stories = stories;
+    this.orderBy = orderBy;
+    this.limit = limit;
+    this.offset = offset;
+  }
+
+  public Map<String, Object> getAsMap() {
+    Map<String, Object> returnValues = new HashMap<>();
+
+    if (name != null) {
+      returnValues.put(CharactersQuery.QUERY_NAME, name);
+    }
+
+    if (nameStartWith != null) {
+      returnValues.put(CharactersQuery.QUERY_NAME_START_WITH, nameStartWith);
+    }
+
+    if (modifiedSince != null) {
+      returnValues.put(CharactersQuery.QUERY_MODIFIED_SINCE, modifiedSince);
+    }
+
+    if (comics != null) {
+      returnValues.put(CharactersQuery.QUERY_COMICS, comics);
+    }
+
+    if (series != null) {
+      returnValues.put(CharactersQuery.QUERY_SERIES, series);
+    }
+
+    if (events != null) {
```

Comment:
By construction, ``comics``, ``series``, ``events`` and ``stories`` won't ever be null